openverse_catalog/dags/providers/provider_csv_load_scripts/inaturalist/05_export_dbpage_to_json_template.sql

```diff
@@ -0,0 +1,92 @@
+/*
+-------------------------------------------------------------------------------
+EXPORT DB PAGE TO JSON
+-------------------------------------------------------------------------------
+
+Please note: This SQL will not run as is! You must replace db_page_number with
+an integer representing the page you want to retrieve.
+
+This takes a single page of data from inaturalist.photos, finds matching
+records from observers, observations, and taxa, and returns a json for
+further processing.
+*/
+
+WITH
+PHOTO_PAGE AS (
+    SELECT
+        PHOTO_ID,
+        EXTENSION,
+        LICENSE,
+        WIDTH,
+        HEIGHT,
+        OBSERVER_ID,
+        OBSERVATION_UUID
+    FROM INATURALIST.PHOTOS
+    -- lifted from
+    -- https://www.citusdata.com/blog/2016/03/30/five-ways-to-paginate/
+    WHERE CTID = ANY(ARRAY(
+            SELECT ('(db_page_number,' || S.I || ')')::tid
+            FROM
+                GENERATE_SERIES(
+                    db_page_number, CURRENT_SETTING('block_size')::int / 4
+                )
+                AS S(I)
+        )
+    )
```

Comment:
Dang, this is really cool! I was about to suggest limit/offset, but that article you linked brings up some good points. I think it might be helpful adding an explicit note here _why_ we aren't using limit/offset for future readers!

Reply:
Thanks. I had been concerned about it taking too long to build the index that would be needed for limit/offset to work. But I was able to test it locally and it didn't take long even on the full dataset. More details [here](https://github.com/WordPress/openverse-catalog/blob/bf452b494466a2c9f1e7ade3bf986e12dce88607/openverse_catalog/dags/providers/provider_csv_load_scripts/inaturalist/01_photos.sql#L40).